openscenario/map_fragment/include/map_fragment/road_segment.hpp

```diff
@@ -0,0 +1,155 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MAP_FRAGMENT__ROAD_SEGMENT__HPP_
+#define MAP_FRAGMENT__ROAD_SEGMENT__HPP_
+
+#include <lanelet2_core/geometry/Lanelet.h>
+#include <map_fragment/map_fragment.hpp>
+#include <map_fragment/geometry.hpp>
+
+namespace map_fragment
+{
+struct RoadCrossSectionDescription
+{
+  const int number_of_lanes;
+  const double lane_width;
+
+  explicit RoadCrossSectionDescription(int number_of_lanes,
+                                       double lane_width)
+  : number_of_lanes(number_of_lanes)
+  , lane_width(lane_width)
+  {
+    if (number_of_lanes <= 0)
+    {
+      throw std::invalid_argument(
+        "Expected number_of_lanes to be positive. Actual value: "
+        + std::to_string(number_of_lanes)
+      );
+    }
```

Comment:
`std::size_t` can be 0, and `number_of_lanes` should be strictly positive. So this check would still be required.

As for the type: I wanted to use `unsigned int`, but then I came across [this excerpt](https://google.github.io/styleguide/cppguide.html) from Google C++ Style Guide:

> You should not use the unsigned integer types such as uint32_t, unless there is a valid reason such as representing a bit pattern rather than a number, or you need defined overflow modulo 2^N. **In particular, do not use unsigned types to say a number will never be negative**. Instead, use assertions for this.

It seemed unusual, but since Autoware theoritically adheres to this guide, I decided to go with it. I wonder what your thoughts are on this particular rule.

Reply:
> It seemed unusual,

I think so too.

After discussing it with some of the more enthusiastic programmers in our company, we concluded that this was nothing more than a religious issue.
Even the Google C++ Style Guide seems to be talking about mathematical integers, the concept of numeric types, and the representation of integer data on real machines without distinguishing between their respective perspectives.
Each has different ideals and practical requirements, and the rules that are correct from one perspective may not be correct from another perspective.

Answer: Follow the Google C++ Style Guide. In other words, you can leave it as `int`. Please ignore the review comments.